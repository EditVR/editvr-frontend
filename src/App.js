/**
 * @file App.js
 * Exports a React component that handles routing.
 */

import React, { Fragment } from 'react';
import { HashRouter as Router, Route } from 'react-router-dom';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import { Home } from './pages';

import muiTheme from './styles/muiTheme';

const App = () => (
  <Fragment>
    <CssBaseline />
    <MuiThemeProvider theme={muiTheme}>
      <Router>
        <Route exact path="/" component={Home} />
      </Router>
    </MuiThemeProvider>
  </Fragment>
);

export default App;
