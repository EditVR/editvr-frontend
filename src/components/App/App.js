/**
 * @file App.js
 * Exports a React component that handles routing.
 */

import React, { Fragment } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { CssBaseline } from '@material-ui/core';

import {
  Home,
  Login,
  Logout,
  Dashboard,
  ExperienceCreate,
  ExperienceEdit
} from '../../pages';
import { PrivateRoute, PublicRoute } from '../../hoc';

const App = () => (
  <Fragment>
    <CssBaseline />
    <Router>
      <Switch>
        <Route exact path="/" component={Home} />
        <PublicRoute
          exact
          path="/login"
          redirectTo="/dashboard"
          component={Login}
        />
        <PrivateRoute exact path="/logout" redirectTo="/" component={Logout} />
        <PrivateRoute
          exact
          path="/dashboard"
          redirectTo="/login"
          component={Dashboard}
        />
        <PrivateRoute
          exact
          path="/experience/create"
          redirectTo="/login"
          component={ExperienceCreate}
        />
        <PrivateRoute
          exact
          path="/experience/edit/:experienceSlug"
          redirectTo="/login"
          component={ExperienceEdit}
        />
      </Switch>
    </Router>
  </Fragment>
);

export default App;
