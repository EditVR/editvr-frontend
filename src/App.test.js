/**
 * @file App.test.js
 * Contains tests for App.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import App from './App';

describe('<App />', () => {
  it('Matches its snapshot', () => {
    expect(renderer.create(<App />).toJSON()).toMatchSnapshot();
  });
});
