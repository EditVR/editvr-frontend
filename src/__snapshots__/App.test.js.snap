// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Matches its snapshot 1`] = `
<div
  className="ThinLayout-wrapper-2"
  id="layout__wrapper"
>
  <div
    className="ThinLayout-thinWrapper-3"
    id="layout__thin-wrapper"
  >
    <header
      className="Header-wrapper-4"
    >
      <img
        alt="EditVR logo"
        className="Header-logo-5"
        src="editvr-logo.svg"
      />
    </header>
    <p
      className="MuiTypography-root-6 MuiTypography-body1-15"
      type="body1"
    >
      <b>
        EditVR is a free, open-source WebVR editor
      </b>
       that lets you produce interactive and affordable 360 tours and VR stories. EditVR makes immersive experiences available to everyone using devices they already own. Nothing to buy, download, or install!
    </p>
    <a
      aria-current={false}
      className={undefined}
      href="#/login"
      onClick={[Function]}
      style={undefined}
    >
      <button
        className="MuiButtonBase-root-46 MuiButton-root-30 MuiButton-raised-36 MuiButton-raisedPrimary-37 Home-button-1"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-31"
        >
          Login
        </span>
        <span
          className="MuiTouchRipple-root-49"
        />
      </button>
    </a>
    <a
      aria-current={false}
      className={undefined}
      href="#/register"
      onClick={[Function]}
      style={undefined}
    >
      <button
        className="MuiButtonBase-root-46 MuiButton-root-30 MuiButton-raised-36 MuiButton-raisedPrimary-37 Home-button-1"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-31"
        >
          Register
        </span>
        <span
          className="MuiTouchRipple-root-49"
        />
      </button>
    </a>
    <footer
      className="Footer-wrapper-56"
    >
      <div
        className="Footer-elements-57"
      >
        <a
          href="https://fourkitchens.com"
        >
          <p
            className="MuiTypography-root-6 MuiTypography-body1-15 Footer-text-59"
            type="body1"
          >
            EditVR is a product of
          </p>
          <img
            alt="Four Kitchens"
            className="Footer-logo-58"
            src="4k-logo-reversed.svg"
          />
        </a>
      </div>
    </footer>
  </div>
</div>
`;
